### PR TITLE
Add bootsnap to improve boot speed

### PR DESCRIPTION
**Pull request by mitchellhenke**
We added bootsnap to the IDP in https://github.com/18F/identity-idp/pull/4017 / https://github.com/18F/identity-idp/pull/5444, and we can add it here to so that we can boot faster.

### CODE REVIEW

**Review by Sgtpluck — APPROVED** [review]

File: Gemfile [hunk 1]
```diff
@@ -15,6 +15,7 @@ gem 'newrelic_rpm', '~> 8.0'
 gem 'pg'
 gem 'pry-rails'
 gem 'puma'
+gem 'bootsnap', '~> 1.0', require: false
```
Comment (zachmargolis) [inline Gemfile:18]: should we add this to a `development` only group?

Comment (mitchellhenke) [reply Gemfile:18]: I'm comfortable running it in production. I've tested it in my sandbox and it's worked ok so far on the IDP side in prod.

Comment (zachmargolis) [reply Gemfile:18]: oh I didn't realize the IDP ran it in prod too!

Comment (mitchellhenke) [reply Gemfile:18]: Yep, as of https://github.com/18F/identity-idp/pull/5444!

Comment (zachmargolis) [reply Gemfile:18]: I love how I even approved that PR and had comments 🤦

**Pull request merged**